pro/src/components/ArchiveConfirmationModal/__specs__/utils.spec.ts

```diff
@@ -0,0 +1,132 @@
+        }),
+      ])
+    ).toStrictEqual(true)
+  })
+
+  it('should return true if an collective offer with status "INACTIVE"', () => {
+    expect(
+      canArchiveCollectiveOffer([
+        collectiveOfferFactory({
+          status: CollectiveOfferStatus.INACTIVE,
+          stocks,
+        }),
+      ])
+    ).toStrictEqual(true)
+  })
+
+  it('should return true if an collective offer with status "EXPIRED" and booking status "CANCELLED"', () => {
+    expect(
+      canArchiveCollectiveOffer([
+        collectiveOfferFactory({
+          status: CollectiveOfferStatus.EXPIRED,
+          stocks,
+          booking: { booking_status: CollectiveBookingStatus.CANCELLED, id: 1 },
+        }),
+      ])
+    ).toStrictEqual(true)
+  })
+
+  it('should return false if an collective offer with status "EXPIRED" and booking status "PENDING"', () => {
+    expect(
+      canArchiveCollectiveOffer([
+        collectiveOfferFactory({
+          status: CollectiveOfferStatus.EXPIRED,
+          stocks,
+          booking: { booking_status: CollectiveBookingStatus.PENDING, id: 1 },
+        }),
+      ])
+    ).toStrictEqual(false)
+  })
+
+  it('should return false if an collective offer with status "SOLD_OUT"', () => {
+    expect(
+      canArchiveCollectiveOffer([
+        collectiveOfferFactory({
+          status: CollectiveOfferStatus.SOLD_OUT,
+          stocks,
+        }),
+      ])
+    ).toStrictEqual(false)
+  })
+
+  it('should return true if an collective offer with status "PENDING" and booking status "REIMBURSED"', () => {
+    expect(
+      canArchiveCollectiveOffer([
+        collectiveOfferFactory({
+          status: CollectiveOfferStatus.PENDING,
+          stocks,
+          booking: {
+            booking_status: CollectiveBookingStatus.REIMBURSED,
+            id: 1,
+          },
+        }),
+      ])
+    ).toStrictEqual(true)
+  })
+
+  it('should return false if an collective offer with status "PENDING" and booking status "USED" and after beginningDatetime + 48h ', () => {
+    expect(
+      canArchiveCollectiveOffer([
+        collectiveOfferFactory({
+          status: CollectiveOfferStatus.PENDING,
+          stocks: [
+            { ...stocks[0], beginningDatetime: String(new Date('2020-01-01')) },
+          ],
+          booking: {
+            booking_status: CollectiveBookingStatus.USED,
+            id: 1,
+          },
+        }),
+      ])
+    ).toStrictEqual(false)
+  })
+
+  it('should return true if an collective offer with status "PENDING" and booking status "USED" and before beginningDatetime + 48h ', () => {
+    expect(
+      canArchiveCollectiveOffer([
+        collectiveOfferFactory({
+          status: CollectiveOfferStatus.PENDING,
+          stocks: [{ ...stocks[0], beginningDatetime: String(new Date()) }],
+          booking: {
+            booking_status: CollectiveBookingStatus.USED,
+            id: 1,
+          },
+        }),
+      ])
+    ).toStrictEqual(true)
+  })
+})
```

Comment:
Pour gagner en concision ça pourrait valoir le coup une prochaine fois d'écrire un test paramétré avec un `it.each` :)